validator_client/src/http_api/mod.rs

```diff
@@ -1254,14 +1251,43 @@ where
                     .body(vec![])
                     .expect("can produce simple response from static values"),
             };
-
-            let body: &Vec<u8> = response.body();
-            let signature = signer(body);
-            let header_value =
-                HeaderValue::from_str(&signature).expect("hash can be encoded as header");
-
-            response.headers_mut().append("Signature", header_value);
-
             response
-        })
+        }
+        Err(e) => {
+            let resp = warp_utils::reject::handle_rejection(e).await;
+            //Building the response with the Rejection
+            match resp {
+                Ok(reply) => {
+                    let response = reply.into_response();
+                    let (_parts, body) = response.into_parts();
+                    let body_bytes = hyper::body::to_bytes(body).await.unwrap();
+                    Response::builder()
+                        .status(StatusCode::BAD_REQUEST)
+                        .body(body_bytes.to_vec())
+                        .expect("can't produce response from rejection")
```

Comment:
I'm not sure if I understand this, why do we convert this into a `BAD_REQUEST` response and throw away the status code from `handle_rejection`?

It looks like we're converting the body from json -> bytes -> vec here, which feels a bit unnecessary.